proton

```diff
@@ -1,511 +1,16 @@
-        f.write("cd \"" + os.getcwd() + "\"\n")
-        f.write("DEF_CMD=(")
-        first = True
-        for arg in sys.argv[2:]:
-            if first:
-                f.write("\"" + arg + "\"")
-                first = False
-            else:
-                f.write(" \"" + arg + "\"")
-        f.write(")\n")
-        dump_dbg_env(f)
-        f.write("\t\"" + wine_path + "\" winedbg --gdb \"${@:-${DEF_CMD[@]}}\"\n")
-    os.chmod(dir + "gdb_run", 0o755)
-
-    with open(dir + "run", "w") as f:
-        f.write("#!/bin/bash\n")
-        f.write("#Run game or given command in environment\n\n")
-        f.write("cd \"" + os.getcwd() + "\"\n")
-        f.write("DEF_CMD=(")
-        first = True
-        for arg in sys.argv[2:]:
-            if first:
-                f.write("\"" + arg + "\"")
-                first = False
-            else:
-                f.write(" \"" + arg + "\"")
-        f.write(")\n")
-        dump_dbg_env(f)
-        if game_arch == ARCH_UNKNOWN:
-            f.write("\t\"" + wine_path + "\" start \"${@:-${DEF_CMD[@]}}\"\n")
-        else:
-            f.write("\t\"" + wine_path + "\" \"${@:-${DEF_CMD[@]}}\"\n")
-    os.chmod(dir + "run", 0o755)
-
-def run():
-    if "PROTON_DUMP_DEBUG_COMMANDS" in env:
-        try:
-            dump_dbg_scripts()
-        except:
-            log("Unable to write debug scripts! " + str(sys.exc_info()[1]))
-    if game_arch == ARCH_UNKNOWN:
-        #probably a batch script or something, hopefully start.exe can handle it
-        run_wine([wine_path, "start", "/unix"] + sys.argv[2:])
-    else:
-        run_wine([wine_path] + sys.argv[2:])
-
-if sys.version_info[0] == 2:
-    binary_stdout = sys.stdout
-elif sys.version_info[0] == 3:
-    binary_stdout = sys.stdout.buffer
-else:
-    raise Exception("Unsupported python version")
-
-#determine mode
-if sys.argv[1] == "run":
-    #start target app
-    run()
-elif sys.argv[1] == "waitforexitandrun":
-    #wait for wineserver to shut down
-    run_wine([bindir + "/wineserver", "-w"])
-    #then run
-    run()
-elif sys.argv[1] == "getcompatpath":
-    #linux -> windows path
-    path = subprocess.check_output([wine_path, "winepath", "-w", sys.argv[2]], env=env, stderr=lfile)
-    binary_stdout.write(path)
-elif sys.argv[1] == "getnativepath":
-    #windows -> linux path
-    path = subprocess.check_output([wine_path, "winepath", sys.argv[2]], env=env, stderr=lfile)
-    binary_stdout.write(path)
-else:
-    log("Need a verb.")
-    sys.exit(1)
-
-sys.exit(0)
+#!/usr/bin/env bash
+# hack for distros which don't set /usr/bin/env python unless py2 is installed
+
+# get current file's dir
+dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+pybins=('python' 'python3' 'python2')
+
+for pybin in "${pybins[@]}"
+do
+    bin="/usr/bin/$pybin"
+    if [ -e $bin ]; then
+        echo "Proton: using $bin"
+        exec $bin "$dir/proton.py" "$@"
+        break
+    fi
+done
```

Comment:
If we reach that part, python isn't installed, or hasn't been found. It would be nice to tell the user about it.